core/src/main/scala/org/apache/spark/util/collection/ExternalAppendOnlyMap.scala

```diff
@@ -509,8 +509,8 @@ class ExternalAppendOnlyMap[K, V, C](
      */
     private def readNextItem(): (K, C) = {
       try {
-        val k = deserializeStream.readKey().asInstanceOf[K]
-        val c = deserializeStream.readValue().asInstanceOf[C]
+        val k = deserializeStream.get.readKey().asInstanceOf[K]
```

Comment:
If developer call `next()` without checking `hasNext` then this will trigger clean up, which is not expected.

Reply:
@jiangxb1987 I think this does not change the original semantic,since it is only call `cleanup` when `EOFException` was thrown.May i missed something?

Reply:
Here may throw NoSuchElementException.I will fix this.Thanks @jiangxb1987